doc/api/modules.md

```diff
@@ -74,9 +74,10 @@ run directly by testing `require.main === module`.
 For a file `foo.js`, this will be `true` if run via `node foo.js`, but
 `false` if run by `require('./foo')`.
 
-Because `module` provides a `filename` property (normally equivalent to
-`__filename`), the entry point of the current application can be obtained
-by checking `require.main.filename`.
+When the entry point is not a CommonJS module, `require.main` is `undefined`,
+and the main module is out of reach. You can try to make assumptions based on
+`process.cwd()` and `process.argv[1]` values to figure out the entry point of
```

Comment:
I am not sure if this is a good recommendation, and the phrasing here ("try to make assumptions") suggests that the reader should not rely on this approach.

Reply:
The reader probably _shouldn’t_ rely on such an approach 😄 though it will likely occur to some people, so we might want to mention it. We could make the hazard less implied, like “You can try to make assumptions based on … but there are cases where those values will not contain the path to the main entry point.”

Reply:
I don't think we should suggest this option at all since it is not technically correct. If the reader wishes to resort to this approach, then that's a design mistake either in node or in the reader's code.

Reply:
It is still valuable to suggest an alternative like this one as Node is not providing yet another straightforward solution. However, we should clarify that it could not work for all cases and pinpoint the caveats of using this approach

Reply:
Isn't the straightforward solution "have separate files for an entry point versus a reusable library, instead of conflating the two purposes into one file"?

Reply:
> as Node is not providing yet another straightforward solution

That doesn't mean that we should recommend or suggest a method that is not known to be reliable or correct.

Reply:
I agree with Tobias, and I think we'd better leave this problem for userland to solve rather than try to document every edge cases – most users won't need to know what was the entry point of the program, and those who do can certainly find an npm package that cover all the edge cases. I've removed the sentence in https://github.com/nodejs/node/pull/41384/commits/d66a3b590ca30efd66ed41bb6a94bb8b3514892a, PTAL.